apps/system/js/remote/simple_app_window_manager.js

```diff
@@ -0,0 +1,77 @@
+/* global BaseModule */
+'use strict';
+
+(function(exports) {
+  function SimpleAppWindowManager() {
+  }
+
+  SimpleAppWindowManager.SERVICES = [
+    'launchApp'
+  ];
+
+  BaseModule.create(SimpleAppWindowManager, {
+    DEBUG: false,
+
+    name: 'SimpleAppWindowManager',
+
+    REGISTERED_EVENTS: ['mozbrowserclose', 'mozbrowsererror'],
+
+    launchApp: function(config) {
+      this.killCurrentApp();
+
+      var app = new BrowserFrame(config);
+      this.REGISTERED_EVENTS.forEach((type) => {
+        app.element.addEventListener(type, this);
+      });
+
+      this.currentApp = app;
+
+      var appWindow = document.createElement('div');
+      appWindow.appendChild(app.element);
+      appWindow.classList.add('appWindow');
+      appWindow.addEventListener('animationend', this);
+      this.container.appendChild(appWindow);
+
+      appWindow.classList.add('opening');
+
+      app.container = appWindow;
+    },
+
+    killCurrentApp: function() {
+      if (!this.currentApp) {
+        return;
+      }
+
+      var app = this.currentApp;
+      this.REGISTERED_EVENTS.forEach((type) => {
+        app.element.removeEventListener(type, this);
+      });
+
+      app.container.removeEventListener('animationend', this);
+      this.container.removeChild(app.container);
```

Comment:
Whenever the DOM element is removed from the document, the event listener should be automatically removed.